src/app/components/queue/form/QuestionForm.tsx

```diff
@@ -0,0 +1,244 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  Drawer,
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  TextField,
+} from "@mui/material";
+import { IoMdClose } from "react-icons/io";
+import React from "react";
+import { officeHourContext } from "@context/OfficeHourContext";
+import { MultipleChoiceTags, SingleChoiceTags } from "./Tags";
+import { TagOption } from "@interfaces/db";
+import { createQuestion } from "@utils/index";
+import { serverTimestamp, Timestamp } from "firebase/firestore";
+import { UserSessionContext } from "@context/UserSessionContext";
+import { IdentifiableQuestion } from "@interfaces/type";
+import FileUploadOutlinedIcon from "@mui/icons-material/FileUploadOutlined";
+
+interface QuestionFormProps {
+  // button to open the form
+  triggerButton: JSX.Element;
+  title: string;
+  // current state of question
+  currentQuestion: IdentifiableQuestion;
+}
+
+const QuestionForm = (props: QuestionFormProps) => {
+  const { triggerButton, title, currentQuestion } = props;
+  const [openForm, setOpenForm] = React.useState(false);
+  const ohContext = React.useContext(officeHourContext);
+  const userSessionContext = React.useContext(UserSessionContext);
+
+  const [question, setQuestion] = React.useState(currentQuestion.title);
+  const [description, setDescription] = React.useState(
+    currentQuestion.description
+  );
+  const [questionPublic, setQuestionPublic] = React.useState(
+    currentQuestion.questionPublic
+  );
+
+  const [questionTags, setQuestionTags] = React.useState<
+    Record<string, TagOption[]>
+  >(() => {
+    const curr_tags = currentQuestion.tags.map((t) => t.choice);
+    let init: Record<string, TagOption[]> = {};
+    Object.keys(ohContext.course.tags).forEach((t) => {
+      init[t] = ohContext.course.tags[t].options.filter((o) =>
+        curr_tags.includes(o.choice)
+      );
+    });
+    return init;
+  });
+
+  const updateQuestionTags = (tagsKey: string, newTags: TagOption[]) => {
+    setQuestionTags({ ...questionTags, [tagsKey]: newTags });
+  };
+
+  const trigger = React.cloneElement(triggerButton, {
+    onClick: () => {
+      setOpenForm(true);
+      if (typeof triggerButton.props.onClick === 'function') {
+        triggerButton.props.onClick();
+      }
+    },
+  });
+
+  const joinQueue = () => {
```

Comment:
I'm not sure what's going on but when I click "Join queue" I'm getting an error saying "Cannot read properties of null (reading 'toDate')". I did some console logging and it seems that the question I just submitted has a timestamp of "null" instead of the current time, which I'm confused about since you literally set the timestamp in this function. Did you ever run into this issue before?

Reply:
But if I reload the page after I get the error, my question shows up and the queue looks great

Reply:
woahh I didn't know about the bug before you found out here. Thanks. It's an onSnapshot bug, fixed it!